include/boost/gil/image_processing/scaling.hpp

```diff
@@ -0,0 +1,115 @@
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/rgb.hpp>
+#include <boost/gil/pixel.hpp>
+#include <boost/gil/image_processing/numeric.hpp>
+
+namespace boost{ namespace gil{
+/// \defgroup ScalingAlgorithms
+/// \brief Algorthims suitable for rescaling
+///
+/// These algorithms are used to improve image
+/// quality after image resizing is made.
+
+/// \defgroup DownScalingAlgorithms
+/// \ingroup ScalingAlgorithms
+/// \brief Algorthims suitable for downscaling
+///
+/// These algorithms provide best results when used
+/// for downscaling. Using for upscaling will probably
+/// provide less than good results.
+
+
+/// \brief a single step of lanczos downscaling
+/// \ingroup DownScalingAlgorithms
+///
+/// Use this algorithm to scale down source image
+/// into a smaller image with reasonable quality.
+/// Do note that having a look at the output once
+/// is a good idea, since it might have ringing
+/// artifacts.
+template <typename ImageView>
+void lanczos_at(
+    long int source_x, 
+    long int source_y, 
+    long int target_x, 
+    long int target_y, 
+    ImageView input_view, 
+    ImageView output_view, 
+    long int a) 
+{
+    using pixel_t = typename std::remove_reference<
+                      decltype(std::declval<ImageView>()(0, 0))
+                    >::type;
+    // C++11 doesn't allow auto in lambdas
+    using channel_t = typename std::remove_reference<
+                        decltype(
+                            std::declval<pixel_t>().at(
+                                std::integral_constant<int, 0>{}
+                            )
+                        )
+                       >::type;
+    pixel_t result_pixel;
+    boost::gil::static_transform(result_pixel, result_pixel, 
+                                 [](channel_t) {
+                                     return static_cast<channel_t>(0);
+                                    });
```

Comment:
style nitpick: this could just read

```
    boost::gil::static_transform(result_pixel, result_pixel, 
        [](channel_t) { return static_cast<channel_t>(0); });
```
without the blank field to the left that makes left-to-right reader's eyes travel before they see anything relevant ;)